src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -118,6 +118,10 @@ def select_interpreter_for_targets(self, targets):
   def _interpreter_from_path(self, path, filters=()):
     try:
       executable = os.readlink(os.path.join(path, 'python'))
+      if not os.path.exists(executable):
+        if os.path.dirname(path) == self._cache_dir:
```

Comment:
This is always `True` in production use from my reading of call-sites in this file. How about re-structuring this private function to allow killing this misleading conditional. Maybe re-name to `_interpreter_from_relpath` and do the `path = os.path.joint(self._cache_dir, relpath)` here allowing unambiguous removal of the conditional.